infrastructure/template.yaml

```diff
@@ -957,6 +957,24 @@ Resources:
                   - !GetAtt OAuthTokenStateMachineRole.Arn
                   - !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-reserved/sso.amazonaws.com/eu-west-2/AWSReservedSSO_di-ipv-cri-otg-hmrc-prod-adm-kms_c1de3f0ff8700ff0"
 
+  DenySecretsIAMPolicy:
+    Condition: IsProdEnvironment
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Deny
+            Action:
+              - "states:GetExecutionHistory"
+            Resource:
+              - !Sub arn:aws:states:${AWS::Region}:${AWS::AccountId}:execution:${AWS::StackName}-OAuthTokenGenerator*
```

Comment:
Can use a `!Ref` here

Reply:
I tested both `!Ref` and `!GetAtt` but neither of them worked and I could still access the execution history via console. I ended up using `!Sub ...::execution::...*` because that was how Visual Editor specified resources for the `GetExecutionHistory` action (admittedly not as good as a direct reference though)

<img width="696" alt="Screenshot 2024-09-24 at 14 09 36" src="https://github.com/user-attachments/assets/becffdbd-6f26-4476-a663-a25642fd6ee0">

<img width="1774" alt="Screenshot 2024-09-24 at 14 02 25" src="https://github.com/user-attachments/assets/16cafb22-1c8b-4218-9d89-b23ffaaab094">